pkg/kubelet/kubelet.go

```diff
@@ -2328,6 +2346,18 @@ func (kl *Kubelet) canAdmitPod(allocatedPods []*v1.Pod, pod *v1.Pod) (bool, stri
 	return true, "", ""
 }
 
+func recordAdmissionRejection(reason string) {
+	// It is possible that the "reason" label can have high cardinality.
+	// To avoid this metric from exploding, we create an allowlist of known
+	// reasons, and only record reasons from this list. Use "Other" reason
+	// for the rest.
+	if admissionRejectionReasons.Has(reason) || strings.HasPrefix(reason, lifecycle.InsufficientResourcePrefix) {
```

Comment:
Is the InsufficientResourcePrefix ever used with extended resources? If so, we probably need to enumerate the non-extended resources, and have a generic "OutOfExtendedResource" for everything else.

Reply:
Done. I enumerated the reason for all non-extended resources in the predicate.go file. PTAL.